packages/buttons/src/elements/Anchor.tsx

```diff
@@ -25,10 +26,19 @@ export const Anchor = forwardRef<HTMLAnchorElement, IAnchorProps>(
       };
     }
 
+    const iconAriaLabel = useText(
+      StyledExternalIcon,
+      { 'aria-label': externalIconLabel },
+      'aria-label',
+      '(opens in a new tab)'
+    );
```

Comment:
```suggestion
    // Prevent useText icon label warning/error unless the anchor is external
    const checkProps = isExternal ? { externalIconLabel } : { noIconLabel: 'true' };
    const iconAriaLabel = useText(
      Anchor,
      checkProps,
      isExternal ? 'externalIconLabel' : 'noIconLabel',
      '(opens in a new tab)'
    );
```

This change ensures the correct warning is logged for the developer ("Warning: you did not provide a customized/translated `externalIconLabel` text value for &lt;Anchor&gt;. Zendesk Garden is rendering &lt;Anchor externalIconLabel="(opens in a new tab)"&gt; by default.") while ensuring the warning/error only occurs if `isExternal` is true.

Reply:
Can do. 👍

Reply:
Sometime in the future, maybe it would be worth it to move this conditional logic into the `useText` helper itself? It's a little confusing (for me, at least!) to see the name of a fake prop inside of the component and realize it's a fake prop/placeholder. 

I'm thinking this could look something like this:

Anchor.tsx
```
const iconAriaLabel = useText(
  Anchor,
  checkProps,
  'externalIconLabel',
  '(opens in a new tab)',
  isExternal
);
```

useText.ts
```
export const useText = (
  component: Pick<FC, 'displayName'>,
  props: Record<string, any>,
  name: string,
  text: string,
  shouldShowWarnings: boolean = true
): string => {
  const value = props[name];

  return useMemo(() => {
    if (shouldShowWarnings) {
      if (name === 'children') {
        // Prevent Garden from providing text as a child content default.
        throw new Error('Error: `children` is not a valid `getText` prop.');
      } else if (value === null || value === '') {
        // Prevent consumer from removing a critical text attribute.
        throw new Error(
          component.displayName
            ? `Error: you must provide a valid \`${name}\` text value for <${component.displayName}>.`
            : `Error: you must provide a valid \`${name}\` text value.`
        );
      } else if (value === undefined) {
        // Warn consumer when Garden's default text is rendered.
        if (process.env.NODE_ENV === 'development') {
          // eslint-disable-next-line no-console
          console.warn(
            component.displayName
              ? `Warning: you did not provide a customized/translated \`${name}\` text value for <${component.displayName}>. Zendesk Garden is rendering <${component.displayName} ${name}="${text}"> by default.`
              : `Warning: you did not provide a customized/translated \`${name}\` text value. Zendesk Garden is rendering ${name}="${text}" by default.`
          );
        }

        return text;
      }

      return value;
    }
  }, [component.displayName, value, name, text, shouldShowWarnings]);
};
```